tests/test_regularization.py

```diff
@@ -0,0 +1,436 @@
+        # cannot be exactly one, as when lambda is scaled down
+        # this brings it to zero.
+        updaters.IntervalParamScaler(1.0, (0.1, 0.9))
+
+    # an interval obviously needs two elements only.
+    with pytest.raises(
+        ValueError,
+        match="tolerable_interval must be a tuple of length 2",
+    ):
+        updaters.IntervalParamScaler(0.5, (0.1, 0.9, 0.5))  # type: ignore
+    with pytest.raises(
+        ValueError,
+        match="tolerable_interval must be a tuple of length 2",
+    ):
+        updaters.IntervalParamScaler(0.5, (0.1,))  # type: ignore
+
+    # the first element of the interval must be at least 0.
+    with pytest.raises(
+        ValueError,
+        match="tolerable_interval must be a tuple whose first element "
+        "is at least 0.*",
+    ):
+        updaters.IntervalParamScaler(0.5, (-0.1, 0.9))
+
+    # the second element of the interval must be greater than the first.
+    with pytest.raises(
+        ValueError,
+        match="tolerable_interval must be a tuple.*the second "
+        "element is greater than the first",
+    ):
+        updaters.IntervalParamScaler(0.5, (0.1, 0.05))
+
+
+@pytest.fixture(scope="module")
+def hierarchical_logger():
+    tmpdir = tempfile.mkdtemp()
+    return imit_logger.configure(tmpdir, ["tensorboard", "stdout", "csv"])
+
+
+@pytest.fixture(scope="module", params=[0.1, 1.0, 10.0])
+def simple_optimizer(request):
+    return th.optim.Adam([th.tensor(request.param, requires_grad=True)], lr=0.1)
+
+
+@pytest.fixture(scope="module", params=[0.1, 1.0, 10.0])
+def initial_lambda(request):
+    return request.param
+
+
+class SimpleRegularizer(regularizers.Regularizer[None]):
+    """A simple regularizer that does nothing."""
+
+    def regularize(self, loss: th.Tensor) -> None:
+        pass  # pragma: no cover
+
+
+@pytest.mark.parametrize(
+    "train_loss",
+    [
+        th.tensor(10.0),
+        th.tensor(1.0),
+        th.tensor(0.1),
+        th.tensor(0.01),
+    ],
+)
+def test_regularizer_update_params(
+    initial_lambda,
+    hierarchical_logger,
+    simple_optimizer,
+    interval_param_scaler,
+    train_loss,
+):
+    regularizer = SimpleRegularizer(
+        initial_lambda=initial_lambda,
+        logger=hierarchical_logger,
+        lambda_updater=interval_param_scaler,
+        optimizer=simple_optimizer,
+    )
+    val_to_train_loss_ratio = interval_param_scaler.tolerable_interval[1] * 2
+    val_loss = train_loss * val_to_train_loss_ratio
+    assert regularizer.lambda_ == initial_lambda
+    assert (
+        hierarchical_logger.default_logger.name_to_value["regularization_lambda"]
+        == initial_lambda
+    )
+    regularizer.update_params(train_loss, val_loss)
+    expected_lambda_value = interval_param_scaler(initial_lambda, train_loss, val_loss)
+    assert regularizer.lambda_ == expected_lambda_value
+    assert expected_lambda_value != initial_lambda
+    # TODO(juan) where's the historic data for this? should check the
```

Comment:
As I think you know at this point, if you don't `dump()` the values just get overwritten. So I think there's nothing to check here...

Reply:
Yeah when I wrote this code I didn't really understand how SB3 loggers worked very well, now I understand them better.